pkg/promutil/promutil.go

```diff
@@ -24,5 +24,9 @@ func MustCounterValue(m prometheus.Metric) float64 {
 	if err := m.Write(&written); err != nil {
 		panic("failed to read Prometheus counter: " + err.Error())
 	}
-	return *written.Counter.Value
+	counter := written.GetCounter()
+	if counter == nil {
+		panic("counter was nil")
+	}
+	return counter.GetValue()
```

Comment:
This is a meaningful change - we were actually depending on this to panic if `Counter` was nil, so the nil-safety fix kinda broke the functionality.